File: frontend/src/types/router.ts
export interface DimensionScore {
  dimension: string;
  score: number;
  weight: number;
  weighted_score: number;
  reason: string;
}

export interface ScoreBreakdown {
  model_id: string;
  model_name: string;
  total_score: number;
  dimension_scores: DimensionScore[];
}

export interface RoutingReason {
  summary: string;
  primary_factors: string[];
  secondary_factors: string[];
  tradeoffs: string[];
}

export interface RiskFlag {
  type: string;
  severity: 'low' | 'medium' | 'high';
  message: string;
  suggestion?: string;
}

export interface RoutingRequest {
  task_id: string;
  goal_id?: string;
  task_type: string;
  task_complexity?: string;
  task_description?: string;
  required_capabilities?: string[];
  preferred_agent_id?: string;
  preferred_model_id?: string;
  context_length_estimate?: number;
  has_vision_input?: boolean;
  requires_tool_calling?: boolean;
  budget_preference?: string;
  speed_preference?: string;
}

export interface RoutingResult {
  selected_model_id: string;
  selected_agent_id?: string;
  backup_model_ids: string[];
  routing_reason: RoutingReason;
  confidence: number;
  risk_flags: RiskFlag[];
  score_breakdown: ScoreBreakdown[];
  is_user_override: boolean;
  override_note?: string;
}

export interface OverrideRequest {
  task_id: string;
  selected_model_id: string;
  original_model_id?: string;
  reason?: string;
}

export interface OverrideResponse {
  task_id: string;
  selected_model_id: string;
  is_user_override: boolean;
  override_note?: string;
}

export interface RoutingRules {
  weights: Record<string, number>;
  role_preferences: Record<string, string[]>;
  hard_constraints: string[];
}

export interface ApiResponse<T> {
  success: boolean;
  data?: T;
  error?: {
    code: string;
    message: string;
  };
}

export const TASK_TYPE_LABELS: Record<string, string> = {
  planning: '规划任务',
  coding: '编码任务',
  review: '审查任务',
  research: '调研任务',
  summarization: '摘要任务',
  supervision: '监督任务',
  debugging: '调试任务',
  documentation: '文档任务',
  testing: '测试任务',
  general: '通用任务',
};

export const DIMENSION_LABELS: Record<string, string> = {
  capability_match: '能力匹配度',
  role_match: '角色匹配度',
  context_fit: '上下文适配度',
  cost_fit: '成本适配度',
  speed_fit: '速度适配度',
  quota_health: '额度健康度',
  historical_performance: '历史表现',
};

export const DIMENSION_COLORS: Record<string, string> = {
  capability_match: 'bg-blue-500',
  role_match: 'bg-green-500',
  context_fit: 'bg-purple-500',
  cost_fit: 'bg-yellow-500',
  speed_fit: 'bg-cyan-500',
  quota_health: 'bg-orange-500',
  historical_performance: 'bg-stone-400',
};

File: backend/src/schemas/router.py
from typing import List, Optional
from pydantic import BaseModel, Field


class DimensionScore(BaseModel):
    dimension: str
    score: float = Field(..., ge=0.0, le=1.0)
    weight: float = Field(..., ge=0.0, le=1.0)
    weighted_score: float
    reason: str


class ScoreBreakdown(BaseModel):
    model_id: str
    model_name: str
    total_score: float
    dimension_scores: List[DimensionScore]


class RoutingReason(BaseModel):
    summary: str
    primary_factors: List[str]
    secondary_factors: List[str]
    tradeoffs: List[str]


class RiskFlag(BaseModel):
    type: str
    severity: str
    message: str
    suggestion: Optional[str] = None


class RoutingRequest(BaseModel):
    task_id: str
    goal_id: Optional[str] = None
    task_type: str = Field(..., description="e.g. planning, coding, review, research, summarization, supervision, debugging, documentation, testing, general")
    task_complexity: str = Field(default="moderate", description="simple | moderate | complex | very_complex")
    task_description: Optional[str] = None
    required_capabilities: Optional[List[str]] = Field(default_factory=list)
    preferred_agent_id: Optional[str] = None
    preferred_model_id: Optional[str] = None
    context_length_estimate: int = Field(default=8000, ge=0)
    has_vision_input: bool = False
    requires_tool_calling: bool = False
    budget_preference: str = Field(default="medium", description="low | medium | high | unlimited")
    speed_preference: str = Field(default="balanced", description="fast | balanced | quality")


class RoutingResult(BaseModel):
    selected_model_id: str
    selected_agent_id: Optional[str] = None
    backup_model_ids: List[str]
    routing_reason: RoutingReason
    confidence: float = Field(..., ge=0.0, le=1.0)
    risk_flags: List[RiskFlag]
    score_breakdown: List[ScoreBreakdown]
    is_user_override: bool = False
    override_note: Optional[str] = None


class OverrideRequest(BaseModel):
    task_id: str
    selected_model_id: str
    original_model_id: Optional[str] = None
    reason: Optional[str] = None


class OverrideResponse(BaseModel):
    task_id: str
    selected_model_id: str
    is_user_override: bool = True
    override_note: Optional[str] = None


class RoutingRulesResponse(BaseModel):
    weights: dict
    role_preferences: dict
    hard_constraints: List[str]


class AgentSelectionRequest(BaseModel):
    task_id: str
    goal_id: Optional[str] = None
    task_type: str = "general"
    risk_level: str = Field("low", pattern="^(low|medium|high)$")
    required_capabilities: List[str] = Field(default_factory=list)
    required_tools: List[str] = Field(default_factory=list)
    workspace_scope: Optional[str] = None
    input_type: str = "text"
    output_type: str = "text"
    context_length_estimate: int = Field(8000, ge=0)
